src/NuGet.Services.BasicSearch/ConfigurationService.cs

```diff
@@ -35,27 +59,40 @@ public bool TryGet(string key, out string value)
                 {
                     var cloudKey = key.Replace(':', '-'); // no ':' supported in cloud services
                     value = SafeRoleEnvironment.GetConfigurationSettingValue(cloudKey);
-                    return true;
+                    setValue = true;
```

Comment:
Are you sure you don't want a
```
if (value != null)
{
    setValue = true;
}
```
here?

Is it impossible for `GetConfigurationSettingValue(cloudKey)` to return null?